### PR TITLE
feat(docs): build out client data fetching stub

### PR DESCRIPTION
**Pull request by amberleyromo**
Build out client data fetching stub.

### CODE REVIEW

File: docs/docs/client-data-fetching.md [hunk 1]
```diff
@@ -2,7 +2,219 @@
 title: Client data fetching
 ---
 
-This is a stub. Help our community expand it.
+Because a Gatsby site hydrates into a React app after loading statically, Gatsby is not just for static sites. You can fetch data dynamically on the client, as needed, as you would with any other React app.
 
-Please use the [Gatsby Style Guide](/docs/gatsby-style-guide/) to ensure your
-pull request gets accepted.
+To illustrate this, we'll walk through a small example site that uses both Gatsby's data layer, and also fetches data on the client. This example is based loosely on Jason Lengstorf's [Gatsby with Apollo](https://github.com/jlengstorf/gatsby-with-apollo) example. We'll be fetching character data for Rick (of Rick and Morty) and a random pupper image.
```
Comment (DSchau) [inline docs/docs/client-data-fetching.md]:
```suggestion
 To illustrate this, we'll walk through a small example site that uses both Gatsby's data layer at build-time and data on the client at run-time. This example is based loosely on Jason Lengstorf's [Gatsby with Apollo](https://github.com/jlengstorf/gatsby-with-apollo) example. We'll be fetching character data for Rick (of Rick and Morty) and a random pupper image.
```

File: docs/docs/client-data-fetching.md [hunk 2]
```diff
@@ -2,7 +2,219 @@
...
+        <p>Rick & Morty API data loads at build time.</p>
+        <p>Dog API data loads at run time.</p> // highlight-line
+        <div>
+          <img
+            src={character.image}
+            alt={character.name}
+            style={{ width: 300 }}
+          />
+        </div>
+        // highlight-start
```
Comment (DSchau) [inline docs/docs/client-data-fetching.md]: Nit, but this is invalid JSX. People sometimes (?) use these docs outside of gatsbyjs.org, so I like to keep these valid, e.g.

```suggestion
         {/* highlight-start */}
```

File: docs/docs/client-data-fetching.md [hunk 3]
```diff
@@ -2,7 +2,219 @@
...
+          ) : img && breed ? (
+            <>
+              <h2>{`${breed} pupper!`}</h2>
+              <img src={img} alt={`cute random `} style={{ maxWidth: 300 }} />
+            </>
+          ) : (
+            <p>Oh noes, error fetching pupper :(</p>
+          )}
+        </div>
+      </div> // highlight-end
```
Comment (DSchau) [inline docs/docs/client-data-fetching.md]:
```suggestion
       </div> {/* highlight-end */}
```

File: docs/docs/client-data-fetching.md [hunk 4]
```diff
@@ -2,7 +2,219 @@
...
+      .catch(error => {
+        this.setState({ loading: false, error })
+      })
+  }
+} // highlight-end
+
+export default ClientFetchingExample
+```
+
+That's it -- an example of querying for data at build time using the Gatsby layer, and dynamically requesting data on the client at run time.
```
Comment (DSchau) [inline docs/docs/client-data-fetching.md]:
```suggestion
 That's it -- an example of querying for data at build time using the Gatsby GraphQL data layer and dynamically requesting data on the client at run time.
```

File: docs/docs/client-data-fetching.md (hunk 2 above)
Comment (amberleyromo) [reply docs/docs/client-data-fetching.md]: Great point, that didn't occur to me at the time.

**Comment by amberleyromo** [comment]
@DSchau unclear why the linting failed?

**Comment by KyleAMathews** [comment]
Probably was a glitch. Anyways, this is good stuff so let's get it in!

**Pull request merged**